src/pixiewps.c

```diff
@@ -462,10 +474,12 @@ int main(int argc, char **argv) {
 
 	uint_fast8_t k = 0;
 	uint_fast8_t found_p_mode = NONE;
-	unsigned int pin;
+	char *pin;
```

Comment:
Being just 8 + 1 characters in length I would avoid using dynamic memory allocation:
```C
char pin[WPS_PIN_LEN + 1];
```